src/nrniv/nrnbbcore_write/io/bbcore_write.h

```diff
@@ -0,0 +1,48 @@
+#ifndef NRN_BBCORE_WRITE_H
+#define NRN_BBCORE_WRITE_H
+
+#include <string>
+#include <vector>
+
+class CellGroup;
+class NrnThread;
+union Datum;
+class NrnMappingInfo;
+
+std::string get_write_path();
+std::string get_filename(const std::string &path, std::string file_name);
+
+// returns start pointer of the container's data
+template<typename T>
+inline T *begin_ptr(std::vector <T> &v) {
+    return v.empty() ? NULL : &v[0];
+}
+
+
+extern "C" {
+
```

Comment:
Just a comment for next refactoring : we don't need all functions with "extern C" linkage. Only callbacks functions are used from CoreNEURON. Rest can remain non-extern-C.